[WIP] - Update guidelines

fgravin opened the pull request:
This PR is a work in progress on additionnal guidelines for ngeo and gmf developpers.
It adds a new file, that will be in the end the final guildeline file (it's gonna be merged with the actual guideline.md i think).
The file can be preview here : https://github.com/fgravin/ngeo/blob/guidelines/docs/guidelines.md#development-guidelines

Also, some parts of the developper-guide were moved and extended to this new guideline, the purpose is to have
- a developper-guide that tells about how to get ngeo running, and how to set up your environnement to code and test
- guidelines that will give recommandation on how to code in ngeo and gmf.

@fredj is also working on this PR.

Please give your feedback on the content and the structure of this document and the global ngeo documentation.

sbrunner commented on docs/developer-guide.md:
```diff
@@ -382,3 +290,92 @@ Services that are objects (rather than numbers, strings, or functions) may need
+
+#### Examples in gh-pages
+
+When you execute the target `gh-pages`, the makefile creates a folder 
```
The target `examples-hosted` to create the `.build/examples-hosted`, to test it locally, and the target `gh-pages` to upload it to GitHub pages ...

sbrunner commented on docs/developer-guide.md:
```diff
@@ -382,3 +290,92 @@ Services that are objects (rather than numbers, strings, or functions) may need
+
+### Unit tests
+
+#### Writhing tests
```
... => Writing

sbrunner commented on docs/gmf-guide.md:
```diff
@@ -0,0 +1,407 @@
+Before starting to code a new feature in `gmf`, you must determine if this feature
+ is 100% `gmf` specific, or if the feature is generic and could be added to `ngeo` 
+ core. You also need to check in `ngeo` core and examples if you don't find
+  anything that could fit with your needs.
```
Extra spaces ?

sbrunner commented on docs/gmf-guide.md:
```diff
@@ -0,0 +1,407 @@
+
+## Templating
+
+### In `ngeo` avoid template when not needed.
```
Shouldn't be in  `docs/developer-guide.md`?

sbrunner commented on docs/gmf-guide.md:
```diff
@@ -0,0 +1,407 @@
+          ngeo.baseTemplateUrl + '/<name>.html';
+    });
+
+ngeo.popupDirective = function(ngeo<Name>TemplateUrl) {
```
popup => <name>

sbrunner commented:
Just some comments otherwise that's looks good :-)

fgravin commented:
I have updated the doc with your comments.
I also removed the old guideline file and merge its content with this file.
I also move from developper-guide.md all that concerns the development.

I've fixed whitespace errors for travis to pass.

fredj commented:
Looks very good to me, thanks a lot @fgravin !

sbrunner commented:
Looks excellent, many thanks
